webots_ros2_core/webots_ros2_core/utils.py

```diff
@@ -19,95 +19,172 @@
+        return path
+
+    return None
+
+
+def __get_webots_home(target_version, condition='ge'):
     """Path to the Webots installation directory."""
-    # search first with the environment variables
-    environVariables = ['ROS2_WEBOTS_HOME', 'WEBOTS_HOME']
-    for variable in environVariables:
-        if variable in os.environ and os.path.isdir(os.environ[variable]) and \
-                get_webots_version(os.environ[variable]) == version:
+    def version_condition(found, target):
+        if target is None:
+            return True
+        if found is None:
+            return False
+        if condition == 'eq':
+            return found == target
+        elif condition == 'ge':
+            return found >= target
+        raise Exception('`condition` can be `eq` or `ge`')
+
+    # Search in the environment variables
+    environment_variables = ['ROS2_WEBOTS_HOME', 'WEBOTS_HOME']
+    for variable in environment_variables:
+        if variable in os.environ and version_condition(WebotsVersion.from_path(os.environ[variable]), target_version):
             os.environ['WEBOTS_HOME'] = os.environ[variable]
             return os.environ[variable]
-    # then using the 'which' command
+
+    # Use the 'which' command (Linux and Mac)
     try:
-        path = os.path.split(os.path.abspath(subprocess.check_output(['which', 'webots'])))[0]
+        where_command = 'where' if sys.platform == 'win32' else 'which'
+        path = os.path.split(os.path.abspath(subprocess.check_output([where_command, 'webots'])))[0]
         path = path.decode('utf-8')
-        if os.path.isdir(path) and get_webots_version(path) == version:
+        if os.path.isdir(path) and version_condition(WebotsVersion.from_path(path), target_version):
             os.environ['WEBOTS_HOME'] = path
             return path
     except subprocess.CalledProcessError:
-        pass  # which not available or Webots not found
-    # finally, look at standard installation pathes
-    pathes = [
-        os.path.join(os.environ['HOME'], '.ros', 'webots' + make_short_version(version), 'webots'),
-        '/usr/local/webots',  # Linux default install
-        '/snap/webots/current/usr/share/webots',  # Linux snap install
-        '/Applications/Webots.app',  # macOS default install
-        'C:\\Program Files\\Webots',  # Windows default install
-        os.getenv('LOCALAPPDATA', '') + '\\Programs\\Webots'  # Windows user install
+        # Webots not found
+        pass
+    except FileNotFoundError:
+        # The `which` command not available
+        pass
+
+    # Look at standard installation pathes
+    paths = [
+        '/usr/local/webots',                                    # Linux default install
+        '/snap/webots/current/usr/share/webots',                # Linux snap install
+        '/Applications/Webots.app',                             # macOS default install
+        r'C:\\Program Files\\Webots',                           # Windows default install
```

Comment:
That looks wrong, it should be either:
```suggestion
        'C:\\Program Files\\Webots',                           # Windows default install
```
or
```suggestion
        r'C:\Program Files\Webots',                           # Windows default install
```

Reply:
It worked in tests, but it makes sense not to combine those two